app/api/v2/identity/sessions.rb

```diff
@@ -136,6 +136,87 @@ def get_user(email)
             error!({ errors: ['identity.session.auth0.invalid_params'] }, 422)
           end
         end
+
+        namespace :switch do
+          desc 'Switch user session',
+               success: { code: 200, message: 'Session was switched' },
+               failure: [
+                 { code: 404, message: 'Record is not found' }
+               ]
+          params do
+            optional :oid,
+                     type: String,
+                     desc: 'Organization OID'
+          end
+          post do
+            user = User.find_by(uid: session[:uid])
+            error!({ errors: ['identity.session.not_found'] }, 404) unless user
+
+            aid = params[:oid]
+            unless aid.nil?
+              # Check account in the organization that user belong to
+              members = Membership.joins('LEFT JOIN organizations ON organizations.id = memberships.organization_id')
+                                  .where(user_id: user.id)
+                                  .select('memberships.*,organizations.name, organizations.organization_id')
+                                  .pluck(:organization_id, :'organizations.name', :'organizations.organization_id')
+                                  .map { |id, name, pid| { id: id, name: name, pid: pid } }
+              error!({ errors: ['identity.member.not_found'] }, 404) if members.nil? || members.length.zero?
+
+              # Check user is barong organization admin or not
+              if members.first[:id].zero?
```

Comment:
There is not only one user with the ability to switch to any organization, it's based on user role and abilities associated to it.

Reply:
You're right. I've updated code based on user role with abilities.